Sources/HermesAppDelegate.m

```diff
@@ -608,34 +623,52 @@ - (IBAction) updateStatusBarIcon:(id)sender {
   [statusItem setMenu:statusBarMenu];
   [statusItem setHighlightMode:YES];
 
-  statusItemImageName = @"";
   [self updateStatusBarIconImage:sender];
 }
 
 - (IBAction) updateStatusBarIconImage:(id)sender {
   if (!PREF_KEY_BOOL(STATUS_BAR_ICON))
     return;
 
-  NSString *imageName = nil;
-  if (PREF_KEY_BOOL(STATUS_BAR_ICON_BNW))
-    imageName = (playback.playing.isPlaying) ? @"Pandora-Menu-Dark-Play" : @"Pandora-Menu-Dark-Pause";
-
-  if (imageName == statusItemImageName)
-    return;
-
   NSImage *icon;
-  if (imageName == nil) {
-    icon = [[NSApp applicationIconImage] copy];
-  } else {
-    icon = [NSImage imageNamed:imageName];
+  if (PREF_KEY_BOOL(STATUS_BAR_ICON_BNW)) {
+    
+    icon = [NSImage
+            imageNamed: (playback.playing.isPlaying) ?
+            @"Pandora-Menu-Dark-Play" : @"Pandora-Menu-Dark-Pause"];
     [icon setTemplate:YES];
+    
+  } else if (PREF_KEY_BOOL(STATUS_BAR_ICON_ALBUM)) {
+    
+    if (playback.playing.isPlaying) {
+      NSData *data = [playback lastImg];
+      icon = (data) ? [[NSImage alloc] initWithData:data] :
+                      [NSImage imageNamed: @"missing-album"];
+    } else
+      icon = [[NSApp applicationIconImage] copy];
+    
+  } else {
+    // Use color application image
+    icon = [[NSApp applicationIconImage] copy];
   }
 
+  // Set image size, then set status bar icon
   NSSize size = {.width = 18, .height = 18};
   [icon setSize:size];
-
   [statusItem setImage:icon];
-  statusItemImageName = imageName;
+  
+  // Optionally show song title in status bar
+  if (PREF_KEY_BOOL(STATUS_BAR_SHOW_SONG)) {
+    NSString *title = playback.playing.playingSong.title;
+    if (title.length > STATUS_BAR_STRING_LENGTH) {
+      title = [[NSString alloc] initWithFormat:@"%@...",
+                [title substringToIndex:STATUS_BAR_STRING_LENGTH]];
+    }
+    [statusItem setTitle:title];
+  
+  }
+  else
+    [statusItem setTitle:@""];
```

Comment:
Setting a status item's title is deprecated. Since this is new code, please use a non-deprecated API.